src/halfpipe/tui/base.py

```diff
@@ -0,0 +1,254 @@
+                #   classes="confirm_warning",
+            )
+        )
+
+    def render(self) -> RenderResult:
+        """Render the header icon.
+
+        Returns:
+            The rendered icon.
+        """
+        return self.icon
+
+
+class MyHeader(Header):
+    def compose(self):
+        yield HeaderTitle()
+        yield HeaderHelpIcon()
+        yield HeaderCloseIcon()
+
+
+class RichImage:
+    """Convert the image to a Rich image."""
+
+    # def __rich_console__(self, console: Console, options) -> RenderResult:
+    #     with Image.open(os.path.join(BASE_DIR, "images/halfpipe_logo_v2.png")) as image:
+    #         pixels = Pixels.from_image(image, resize=(110, 92))  # 105, 92
+    #     return console.render(pixels)
+
+
+class ImageContainer(Container):
+    """Create a container for the image."""
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+    def render(self):
+        return RichImage()
+
+    def key_escape(self):
+        self.dismiss(False)
+
+
+class Welcome(ModalScreen):
+    """Intro screen with an intro image."""
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+    def compose(self) -> ComposeResult:
+        yield ImageContainer(id="welcome_image")
+
+    def key_escape(self):
+        self.dismiss(True)
+
+    def on_click(self, event: events.Click) -> None:
+        self.dismiss(True)
+
+
+class MainApp(App):
+    CSS_PATH = [
+        os.path.join(BASE_DIR, "tcss/base.tcss"),
+        os.path.join(BASE_DIR, "feature_widgets/tcss/base.tcss"),
+        os.path.join(BASE_DIR, "feature_widgets/tcss/taskbased.tcss"),
+        os.path.join(BASE_DIR, "feature_widgets/tcss/model_conditions_and_contrasts.tcss"),
+        os.path.join(BASE_DIR, "utils/tcss/file_browser.tcss"),
+        os.path.join(BASE_DIR, "working_directory/tcss/working_directory.tcss"),
+        os.path.join(BASE_DIR, "data_input/tcss/data_input.tcss"),
+        os.path.join(BASE_DIR, "preprocessing/tcss/preprocessing.tcss"),
+        os.path.join(BASE_DIR, "run/tcss/run.tcss"),
+        os.path.join(BASE_DIR, "utils/tcss/path_pattern_builder.tcss"),
+        os.path.join(BASE_DIR, "dev.tcss"),
+        os.path.join(BASE_DIR, "utils/tcss/radio_set_changed.tcss"),
+    ]
+
+    # TODO: The non active tabs should not show the bindings.
+    BINDINGS = [
+        ("w", "show_tab('work_dir_tab')", "Working directory"),
+        ("i", "show_tab('input_data_tab')", "Input data"),
+        ("f", "show_tab('feature_selection_tab')", "Features"),
+        ("p", "show_tab('preprocessing_tab')", "General preprocessing settings"),
+        ("g", "show_tab('models_tab')", "Group level models"),
+        ("r", "show_tab('run_tab')", "Check and run"),
+    ]
+
+    BINDINGS = BINDINGS + [("d", "toggle_dark", "Toggle dark mode")]
+
+    # maybe rename to available_tasks? this is a top level class variable that contains available tasks.
+    available_images: dict = {}
```

Comment:
I think it would be good to discuss how we handle state in the new UI. If there are variables besides the context object where state is kept, then we may want to think about moving them to the context object.